src/Type/ExponentiateHelper.php

```diff
@@ -77,16 +66,37 @@ public static function exponentiate(Type $base, Type $exponent): Type
 		]);
 	}
 
-	private static function exponentiateConstantScalar(ConstantScalarType $base, Type $exponent): ?Type
+	private static function exponentiateConstantScalar(Type $base, Type $exponent): ?Type
 	{
+		$allowedOperandTypes = new UnionType([
+			new IntegerType(),
+			new FloatType(),
+			new IntersectionType([
+				new StringType(),
+				new AccessoryNumericStringType(),
+			]),
+			new BooleanType(),
+			new NullType(),
+		]);
+		if (!$allowedOperandTypes->isSuperTypeOf($exponent)->yes()) {
+			return new ErrorType();
```

Comment:
I disagree with returning ErrorType here.
`string` might still be a `numeric-string` so it's wrong to return ErrorType for that case.

We just want to prevent the internal error the issue is about.